quinn/src/lib.rs

```diff
@@ -304,10 +304,14 @@ impl<'a> EndpointBuilder<'a> {
         Ok(self)
     }
 
-    pub fn set_protocols(&mut self, protocols: &[String]) -> &mut Self {
+    pub fn set_protocols(&mut self, protocols: &[Vec<u8>]) -> &mut Self {
```

Comment:
We should probably take `&[&[u8]]` here instead, to prevent having more allocations than necessary.